appserver/web/web-core/src/main/java/org/apache/catalina/core/ApplicationDispatcher.java

```diff
@@ -1036,12 +1036,6 @@ private ServletRequest wrapRequest(State state) {
             }
             //END OF github/javaee/glassfish/issues/21846
 
-            //START OF github/javaee/glassfish/issues/22079
-            if (DispatcherType.ASYNC.equals(state.dispatcherType)) {
```

Comment:
Did you move that or does it work even without that?

Reply:
It MUST work without that. That is the reverting of a previous special case that should never have been a special case. See https://github.com/eclipse-ee4j/glassfish/issues/22079